shell/platform/darwin/ios/framework/Source/FlutterView.mm

```diff
@@ -49,6 +36,23 @@ - (instancetype)initWithCoder:(NSCoder*)aDecoder {
   return nil;
 }
 
+- (BOOL)isWideGamutSupported {
+#if TARGET_OS_SIMULATOR
+  // As of Xcode 14.1, the wide gamut surface pixel formats are not supported by
+  // the simulator.
+  return NO;
+#endif
+
+  if (![_delegate isUsingImpeller]) {
+    return NO;
+  }
+
+  // This predicates the decision on the capabilities of the iOS device's
+  // display.  This means external displays will not support wide gamut if the
+  // device's display doesn't support it.  It practice that should be never.
+  return UIScreen.mainScreen.traitCollection.displayGamut != UIDisplayGamutSRGB;
```

Comment:
nit: This should be using availability checks and then checking the `UIScene` API on newer iOS and falling back to this deprecated property. That would also solve the problem of not knowing whether a connected display is wide gamut or not.

Reply:
Filed issue: https://github.com/flutter/flutter/issues/125496

Reply:
Can you instead use `self.window.windowScene.screen.traitCollection.displayGamut`? See `mainScreenIfViewLoaded` in the FlutterViewController for a fallback example.

Reply:
Done